base/server/python/pki/server/subsystem.py

```diff
@@ -1173,19 +1173,26 @@ def request_range(self, master_url, range_type, session_id=None, install_token=N
 
     def request_ranges(self, master_url, session_id=None, install_token=None):
 
-        logger.info('Requesting request ID range')
+        if self.config.get('dbs.request.id.generator') == 'legacy':
```

Comment:
In one of the previous iterations this was:
`if self.config.get('dbs.request.id.generator', 'legacy') == 'legacy':`
...i.e. it defaulted to using the sequential serial numbers. I'm wondering why you removed this? It seems to me that without the default this will break existing configs as they will not define this property and therefore won't do any of the `Range` calculations despite being legacy sequential systems.

Reply:
The problem is actually with this line:
```
if self.config.get('dbs.cert.id.generator', 'legacy') == 'legacy' or \
    self.config.get('dbs.key.id.generator', 'legacy') == 'legacy':
```
Even if we have `dbs.cert.id.generator=random` in CA there wouldn't be a `dbs.key.id.generator` in CA so the second part will evaluate to `true`. Similar issue happens in KRA too.

So the options are:
* use the same params for cert ID and key ID (e.g. `dbs.serial.id.generator`) which won't be very descriptive
* add qualifiers into the expression: `if subsystem.type == 'CA' and self.config.get('dbs.cert.id.generator', 'legacy') == 'legacy' or ...` which will make it a bit longer
* create an upgrade script to add `dbs.cert.id.generator=legacy` if it doesn't exist yet, which will increase the size of `CS.cfg`.

Let me try the option #2.